tests/sles4sap/trento/clean_az.pm

```diff
@@ -0,0 +1,26 @@
+use Mojo::Base 'publiccloud::basetest';
+use base 'consoletest';
+use strict;
+use testapi;
+
+use constant TRENTO_AZ_PREFIX => 'openqa-trento';
+use constant TRENTO_AZ_ACR_PREFIX => 'openqatrentoacr';
+
+sub run {
+    my ($self, $args) = @_;
+    $self->select_serial_terminal;
+
+    ######################
+    # az login
+    set_var 'PUBLIC_CLOUD_PROVIDER' => 'AZURE';
+    my $provider = $self->provider_factory();
+    assert_script_run('az group list --query "[].name" -o tsv');
+
+    # look for all ACR and delete resource groups that they belong to
+    assert_script_run('for g in $(for l in $(az acr list --query "[].loginServer" -o tsv | grep ' . TRENTO_AZ_ACR_PREFIX . '); do az acr show --name ${l} --query resourceGroup -o tsv; done); do az group delete --name ${g} -y; done', 3600);
+
+    # check again at the end
+    assert_script_run('az group list --query "[].name" -o tsv | grep ' . TRENTO_AZ_PREFIX);
```

Comment:
`I suppose the whole purpose of the command is to remove Azure resource groups that match the prefixes defined as constants at the top of the module.`
yes definitely

`If that's the case, then wouldn't the expected output of this command be an empty list? Because that would mean that the grep would return with a retval that's not 0, and make the assert_script_run() command fail.`

right, wrong code. the meaning of this **last check** is about **trust in az delete** ... but again definitely the overall point is that this file should not be in this PR

Reply:
Perfect.

BTW, you can quote text with `>`:

> right, wrong code. the meaning of this last check is about trust in az delete ... but again definitely the overall point is that this file should not be in this PR